src/Illuminate/Auth/EloquentUserProvider.php

```diff
@@ -102,19 +102,21 @@ public function updateRememberToken(UserContract $user, $token)
      */
     public function retrieveByCredentials(array $credentials)
     {
+        $model = $this->createModel();
+
         if (empty($credentials) ||
            (count($credentials) === 1 &&
-            array_key_exists('password', $credentials))) {
+            array_key_exists($model->getAuthPasswordField(), $credentials))) {
```

Comment:
This is wrong. 

This basically allow you to do:

```php
Auth::attempt(['username' => 'laravel, 'password' => 'hello']);
``` 

It use a reserved key `password`, but based basically the key is irrelevant since it would get the right value from `User::getAuthPassword()` helper method.

Reply:
This would just check if the field name for password exist, before this it was using static 'password' field. With the combination of `password()` method in `AuthenticatesUsers` and `getAuthPasswordField()` in `Authenticatable` we can customize this field to be different from static password.